### PR TITLE
Experimental:  Provide UBI based community build tool image and dev environment

### PR DESCRIPTION
**Pull request by yxun**
This PR provides an alternative build and development environment using tools such as Podman.
And builds an istio-dev environment with a [Red Hat Universal Base Image (UBI)](https://catalog.redhat.com/software/base-images)
Currently, we can try it as an experimental local environment setup until its getting stable.

More information and instructions are described in  [docker/istio-dev/README.ubi9.md](https://github.com/istio/tools/compare/master...yxun:tools:ossm-ubi9-podman?expand=1#diff-3ac137820ce6dcffdaf0c403d9002c6eca3fa77afba0466c3a8849fdd2729f29)

This PR covers examples of Istio build and test with the new builder image. 
The proxy final image has not been tested. I will update and adjust the proxy build final image in another PR later.

Let me know if there is any concern or improvement suggestion. Thank you.

Gaps between `build-tools/Dockerfile` and `build-tools/Dockerfile.ubi9`:
- This PR `build-tools/Dockerfile.ubi9` does not include the following packages or steps:
   - Ruby: html-proofer, licensee
   - wireshark-cli (tshark)
   - proxy-tsan-instrumented-libcxx

### CODE REVIEW

File: docker/istio-dev/README.ubi9.md [hunk 1]
```diff
@@ -0,0 +1,205 @@
+# Istio Development Environment in Rootless Podman
```
Comment (howardjohn) [inline docker/istio-dev/README.ubi9.md:1]: istio-dev is totally unmaintained and probably broken. I was thinking of submitting a PR to remove it entirely. Is there value? Are you using it?

File: docker/build-tools/Dockerfile.ubi9 [hunk 2]
```diff
@@ -0,0 +1,923 @@
...
+# The container is built using different contexts, one per execution environment (plain binaries, Ruby, nodejs), and
+# then combined into the final image.
+#
+# We pin versions of stuff we install. Modify the various XXX_VERSION variables within the individual build contexts
+# in order to control these versions.
+
+################
+# Binary tools
+################
+ARG BASE_IMAGE=registry.access.redhat.com/ubi9/ubi:9.2-696
```
Comment (howardjohn) [inline docker/build-tools/Dockerfile.ubi9:28]: I don't think we should do this. It doesn't benefit the project and adds high maintenance cost

**Review by howardjohn — CHANGES_REQUESTED** [review]

File: docker/istio-dev/README.ubi9.md (hunk 1 above)
Comment (yxun) [reply docker/istio-dev/README.ubi9.md:1]: I am not using it. I realized that was out of date in the middle of my work on this PR.
I tried to start from a simple KIND setup. I can sync up istio repo kind-provisoiner later.

+1 I agree it's better removing istio-dev deprecated directory.

File: docker/build-tools/Dockerfile.ubi9 (hunk 2 above)
Comment (yxun) [reply docker/build-tools/Dockerfile.ubi9:28]: I see. yes, it adds high maintenance cost. I can mark this PR as a draft and and answer your question about any value later. Currently, this is an experimental setup and I'd like to collect some feedback of this alternative base.

**Pull request closed without merging**